RHIDP-7975 Enabling authentication procedures with mandatory steps only (#1266)
redhat-developer/red-hat-developers-documentation-rhdh: modules/authentication/proc-enabling-user-authentication-with-rhbk-with-mandatory-steps-only.adoc

[id="enabling-user-authentication-with-rhbk-with-mandatory-steps-only"]
= Enabling user authentication with {rhbk-brand-name} ({rhbk}) (with mandatory steps only)

To authenticate users with {rhbk-brand-name} ({rhbk}), enable and configure the OpenID Connect (OIDC) authentication provider in {product} and provision the users and groups from {rhbk} to the {product-short} software catalog.

.Prerequisites
* You link:{configuring-book-url}[added a custom {product-short} application configuration], and have sufficient permissions to modify it.
* You have sufficient permissions in {rhsso} to create and manage a realm and a client.
Alternatively, your {rhbk} administrator can prepare in {rhbk} the required realm and client for you.

.Procedure

> . To allow {product-short} to authenticate with {rhbk}, complete the steps in {rhbk}, to link:https://docs.redhat.com/en/documentation/red_hat_build_of_keycloak/26.0/html/getting_started_guide/getting-started-zip-#getting-started-zip-create-a-realm[create a realm and a user] and link:https://docs.redhat.com/en/documentation/red_hat_build_of_keycloak/26.0/html/getting_started_guide/getting-started-zip-#getting-started-zip-secure-the-first-application[secure the first application]:

Reviewer: I found this section is somewhat wordy and difficult to follow. I understand that the section titles are from Keycloak docs, but I don't think we need to use them as is.

I think we can break down this section into these steps:

This procedure guides you through configuring RHBK to handle authentication for RHDH. It involves setting up a realm, registering your Developer Hub instance as a client, and preparing a user for verification.
1. Set Up the RHBK Realm
2. Register Developer Hub as a Client in RHBK
3. Prepare a User for Verification

Author: ```suggestion
. To allow {product-short} to authenticate with {rhbk}, complete the steps in {rhbk}:
.. link:https://docs.redhat.com/en/documentation/red_hat_build_of_keycloak/26.0/html/getting_started_guide/getting-started-zip-#getting-started-zip-create-a-realm[Create a realm].
link:https://docs.redhat.com/en/documentation/red_hat_build_of_keycloak/26.0/html/getting_started_guide/getting-started-zip-#getting-started-zip-create-a-realm[Create a user for verification].
.. link:https://docs.redhat.com/en/documentation/red_hat_build_of_keycloak/26.0/html/getting_started_guide/getting-started-zip-#getting-started-zip-secure-the-first-application[Register {product-short} as a client in {rhbk}].
```
